core/src/main/java/io/undertow/server/handlers/CookieImpl.java

```diff
@@ -163,20 +164,14 @@ public String getSameSiteMode() {
     }
 
     @Override
-    public Cookie setSameSiteMode(final String sameSiteMode) {
-        if (sameSiteMode != null) {
-            switch (sameSiteMode.toLowerCase(Locale.ENGLISH)) {
-                case "strict":
-                    this.setSameSite(true);
-                    this.sameSiteMode = "Strict";
-                    break;
-                case "lax":
-                    this.setSameSite(true);
-                    this.sameSiteMode = "Lax";
-                    break;
-                default:
-                    throw UndertowMessages.MESSAGES.invalidSameSiteMode(sameSiteMode);
-            }
+    public Cookie setSameSiteMode(final String mode) {
+        final String m = Cookies.lookupSameSiteMode(mode);
+        if (m != null) {
+            UndertowLogger.REQUEST_LOGGER.tracef("Setting SameSite mode to [%s] for cookie [%s]", m, this.name);
+            this.sameSiteMode = m;
+            this.setSameSite(true);
+        } else {
+            UndertowLogger.REQUEST_LOGGER.warnf(UndertowMessages.MESSAGES.invalidSameSiteMode(mode, Arrays.toString(Cookie.SameSiteMode.values())), "Ignoring specified SameSite mode [%s] for cookie [%s]", mode, this.name);
```

Comment:
If just throwing the exception here, a request processing always fails with "500 Internal Server Error" when an invalid SameSite mode is specified. 
I think it's better to just ignore such an invalid SameSite mode (= wrong setting like a typo) and output the WARN log message instead of failing the request processing.

Reply:
Okay, I see your point